src/User/Service/TwoFactorSmsCodeGeneratorService.php

```diff
@@ -67,13 +67,19 @@ public function run()
         
         if( !(null===$mobilePhone) && $mobilePhone!='' ){
```

Comment:
Instead of wrapping all the code logic into an `if`, do the opposite check (number is invalid or not provided) and return `false` immediately, so the rest of the code can be outside a condition